src/kolibri_gnome/desktop_launcher/application.py

```diff
@@ -491,3 +695,100 @@ def open_kolibri_home(self):
+
+
+class ChannelApplication(Application):
+    def __init__(self, channel_id, *args, **kwargs):
+        self.__channel_id = channel_id
+        super().__init__(*args, **kwargs)
+
+    @property
+    def channel_id(self):
+        return self.__channel_id
+
+    @property
+    def default_url(self):
+        return "x-kolibri-app:/learn#topics/{channel_id}".format(
+            channel_id=self.channel_id
+        )
+
+    def _create_window(self, target_url=None):
+        return KolibriChannelWindow(self.channel_id, target_url, delegate=self)
+
+    def url_is_valid(self, url):
+        return super().url_is_valid(url) and self.__is_url_in_channel(url)
+
+    def open_url_in_external(self, url):
+        if self.is_kolibri_url(url):
+            url = self.url_to_x_kolibri_app(url)
+        self.open_in_browser(url)
+
+    def __is_url_in_channel(self, url):
+        url_tuple = urlsplit(url)
+
+        if re.match(r"^\/(?P<lang>\w+\/)?learn\/?", url_tuple.path):
+            return self.__is_learn_fragment_in_channel(url_tuple.fragment)
+        elif re.match(r"^\/(?P<lang>\w+\/)?user\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/(?P<lang>\w+\/)?logout\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/(?P<lang>\w+\/)?redirectuser\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/zipcontent\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/app\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/static\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/downloadcontent\/?", url_tuple.path):
+            return True
+        elif re.match(r"^\/content\/storage\/?", url_tuple.path):
+            return True
+        else:
+            return False
+
+    def __is_learn_fragment_in_channel(self, fragment):
+        fragment = fragment.lstrip("/")
+
+        if re.match(r"^content-unavailable", fragment):
+            return True
+
+        contentnode_id = self.__contentnode_id_for_learn_fragment(fragment)
+
+        if contentnode_id is None:
+            return False
+
+        if contentnode_id == self.channel_id:
+            return True
+
+        response = self.kolibri_api_get(
+            "/api/content/contentnode/{contentnode_id}".format(
+                contentnode_id=contentnode_id
+            )
+        )
+
+        if response is None:
+            return False
+
+        contentnode_channel = response.get("channel_id")
+
+        return contentnode_channel == self.channel_id
+
+    def __contentnode_id_for_learn_fragment(self, fragment):
+        patterns = (
+            r"^topics\/c\/(?P<node_id>\w+)",
+            r"^topics\/t\/(?P<node_id>\w+)",
+            r"^topics\/(?P<node_id>\w+)",
+        )
+
+        for pattern in patterns:
+            match = re.match(pattern, fragment)
+            if match:
+                return match.group("node_id")
```

Comment:
This is something that could also be simplified using just one regular expression:
```suggestion
        pattern = r"^topics\/([ct]\/)?(?P<node_id>\w+)"
        match = re.match(pattern, fragment)
        if match:
            return match.group("node_id")
```